### ISSUE TITLE
Pluto - Automatic minor update

### ISSUE THREAD

**Issue by fvankrieken** [NYCPlanning/data-engineering#97]
Just wanted a place to get down some brainstorming, nothing super concrete at the moment, but thoughts around what would be needed to automatically build pluto minor when new zoning data land in digital ocean. My understanding is that the specific trigger would be GIS running the data library step for zoning data? Or potentially just when they update the source data, and whatever  automated process we have could both trigger archival with data library and then run pluto build.

On the triggering side, a simple first take could be a scheduled action which checks date of update of either data library outputs or the inputs that GIS produces and compares to last PLUTO build date, and if it's more recent than the last build to trigger (data library step and) a build. 

For the build part, we would need to automate the parts right now that come from code currently
- current build version - this can just be last build (major or minor) incremented
- versions of inputs to keep constant. Currently this is somewhat manually updated for the first minor build in a cycle. It should be relatively straightforward to grab `source_data_versions.csv` and get them from there. 
 
`main/latest` would probably be the easiest way to consistently grab those - don't need to iterate through foldernames to find "max" version, just look in main/latest and grab version name from there.

Any other big complications?

@AmandaDoyle @damonmcc @alexrichey

**Comment by alexrichey** [comment]
Just noting the two open questions here, if someone wants to carry this on while I'm out.
1.  We would ideally be notified when a minor version is built. Potentially options: 
  - create a Github Issue 
  - email
  - message on Teams
2.  Published Versions (ie products on edm-publishing) don't have a consistent identifier for their version. With pluto specifically, there's a `version.txt` file in the zipped output. But if we're going to infer versions, we'd want a more efficient way than unzipping a large file just to parse a teeny version text file. 

Regarding the trigger for the build itself, there's a DO shortcoming in that (unlike AWS) you can't trigger off changes to spaces/buckets. So kicking off a build after the Zoning Dataloading job seems like the best option for now.

cc @AmandaDoyle @fvankrieken @damonmcc

**Comment by alexrichey** [comment]
FYI, @AmandaDoyle & @fvankrieken - @jackrosacker had a [comment](https://github.com/NYCPlanning/data-engineering/commit/137239d3408092815ec3be8efac51b604ef182f3#commitcomment-127149520) about this work. Let's discuss when Amanda is back.
```
Hey @alexrichey - I've seen some notifications that indicate a possible trigger from Zoning data upload to start a PLUTO build. Wanted to just note that when we run the ZTLDB Dataloading action, the next step for GIS is to QA the zoning data, which is a prereq for the PLUTO build, and a step that we often have to repeat before giving the green light on PLUTO. I may be missing what you all are working on, but figured it was worth discussing
```

**Comment by AmandaDoyle** [comment]
Hi @jackrosacker - addressing your question about autobuilding ZTLDB and PLUTO minor (and happy to talk anything through if easiest).  When kicking off a build for PLUTO minor or ZTLDB was manual, it definitely made sense to have checks between building ZTLDB and PLUTO minor.  If we automate the initiation of a build to kick off when new data are available, then the manual aspect is removed, and QA just needs to be done.  Therefore, a new process could be:
1) New zoning data are uploaded to DO
2) Builds are initiated for ZTLDB and PLUTO minor
3) QA is done for ZTLDB.
3.a) Issue is found in ZTLDB w/ zoning data and reported.  Cycle restarts.
4) QA is done for PLUTO minor
5) ZTLDB and PLUTO minor are published with monthly data release
The one issue we'll need to be mindful of is computing and storage resources.  If we end up building a ZTLDB and PLUTO minor several times a month and storing all outputs there may be negative impacts on resources/organization and we'll need to do cleanup.

**Comment by jackrosacker** [comment]
Hey @AmandaDoyle - As long as there are no issues with the PLUTO build being triggered potentially a few times a month, the sequence you wrote out above makes a lot of sense to me. 

I would say that typically we only have to build ZTLDB once a month, but maybe every third month we find zoning issues and have to run the build a second time. I think we've only had to run the build 3x in a month once since I've been here (for a sense of scale.)

It would be nice to discuss before this is implemented in production so we can understand how else it may impact the GIS Team process. Tagging @croswell81 and @caseysmithpgh for FYIs and in case I got something wrong.
